### PR TITLE
Zeus cleanup

### PR DESCRIPTION
**Pull request by rpelisse**
@spyrkob Following the integration of Vault, I did a few cleanups. Nothing big, but note that reviewing by commit might be easier than with files changes.

### CODE REVIEW

File: roles/monit/defaults/main.yml [hunk 1]
```diff
@@ -34,4 +34,4 @@ monit_alerts_custom_scripts:
   - { name: "root-podman-too-many-images", src: 'podman-too-many-images.sh.j2', dest: "{{ monit_custom_scripts_directory }}/root-{{ max_nb_images_for_root }}-podman-too-many-images.sh" }
   - { name: "yum-update", src: 'yum-update.sh.j2', dest: "{{ monit_custom_scripts_directory }}/yum-update" }
   - { name: "needs-restarting", src: 'dnf-needs-restarting.sh.j2', dest: "{{ monit_custom_scripts_directory }}/needs-restarting" }
-  - { name: "daily-zeus", src: 'zeus_daily_run_check.sh.j2', dest: "{{ monit_custom_scripts_directory }}/zeus_daily_run_check.sh" }
+#  - { name: "daily-zeus", src: 'zeus_daily_run_check.sh.j2', dest: "{{ monit_custom_scripts_directory }}/zeus_daily_run_check.sh" }
```
Comment (spyrkob) [inline roles/monit/defaults/main.yml:37]: @rpelisse is that disabled due to vault integration or is there another reason?

File: roles/vault/tasks/main.yml [hunk 2]
```diff
@@ -1,10 +1,21 @@
 ---
-- ansible.builtin.assert:
+- name: "Ensure required parameters are provided."
+  ansible.builtin.assert:
     that:
       - vault_paths is defined
       - vault_paths | length > 0
       - ansible_hashi_vault_addr is defined and ansible_hashi_vault_addr | length > 0
+      - ansible_hashi_vault_token is defined and ansible_hashi_vault_token | length > 0
```
Comment (spyrkob) [inline roles/vault/tasks/main.yml:8]: @rpelisse I think that's not correct, `ansible_hashi_vault_token` is optional if `ansible_hashi_vault_secret_id` is present. Or possibly they should even be exclusive?

File: zeus.yml [hunk 3]
```diff
@@ -22,18 +22,25 @@
       - vars/pgsql.yml
       - vars/hosts.yml
       - vars/qualys.yml
+  roles:
```
[the remark this answers is not in the document]
Comment (rpelisse) [reply zeus.yml:25]: @spyrkob Yes, that the default behavior of Ansible (runs roles after `pre_tasks:` and before `tasks:`) so it was strange to have a for loop to include roles in tasks. 

I also think we may have a bit overdo it by moving so many roles in pre_tasks... They would probably be fine being in this list, but firsts.

File: roles/monit/defaults/main.yml (hunk 1 above)
Comment (rpelisse) [reply roles/monit/defaults/main.yml:37]: Vault...

File: roles/vault/tasks/main.yml (hunk 2 above)
Comment (rpelisse) [reply roles/vault/tasks/main.yml:8]: Maybe, but for now, Zeus only use token, so I find better to have an error message remind you to set the TOKEN. If we switch to approle, I'll change that back.

Comment (gaol) [reply roles/vault/tasks/main.yml:8]: yes, they should be exclusive. Either using `token` authentication or `approle` authentication, but not both at the same time.

**Review by spyrkob — APPROVED** [review]
LGTM

**Pull request merged**